### ISSUE TITLE
Feature request: MonadReader instances

### ISSUE THREAD

**Issue by matt19234** [scotty-web/scotty#219]
Is there any way instances could be added for `MonadReader r m => MonadReader r (ScottyT e m)` and `MonadReader r m => MonadReader r (ActionT e m)`? I'm not sure if this is possible, but if it is, I think a lot of users could benefit from this.

In my case, I'm trying add the reader monad to my workflow as follows.

```
type ScottyRM = ScottyT SomeErrorType (ReaderT SomeConfigType IO)
type ActionRM = ActionT SomeErrorType (ReaderT SomeConfigType IO)
```

Ideally, I would like to be able to write functions that work for both of these with a `MonadReader Config` constraint but unfortunately there is no instance for either of them. Moreover, there is no `MonadTrans` instance for `ScottyT e` so I can't use `lift` either.

**Comment by maksbotan** [comment]
This should be fixed by #232 (merged).

**Comment by chessai** [comment]
yep, thanks @maksbotan !
